eslint config added, one minute timer token expiration timer added

mat888 opened the pull request:
*Please ignore my two previous pull requests. Apologies for any confusion or clutter.

In index.js, the while loop has been replaced by a function `checkToken()` which is called by `setInterval()` every 1 second. `setInterval()` is terminated when `tokenTimer` reaches 60 seconds, as well as when max attempts or successful verification is achieved.

A eslint config was added. Despite choosing Node, it shows `"browser": true` in the file, not sure if this should be manually changed or if this is intended.

mat888 committed:
token timer and cleanup

mat888 committed:
airbnb eslint config added

mat888 committed:
fixed expiration time units (ms -> s)

mat888 committed:
fixed comment

MatusMak commented:
Hey, thanks for PR!

So just to make it clear:

- this addresses #1 and #2?
- can your previous two PR be closed and should we go forward only with this one?

I missed notifications about your PRs, sorry about that. I will get back to you tomorrow or the day after.

mat888 commented:
Hey, sorry for any confusion.

Correct, this addresses issues #1 and #2 - the other two PRs were the result of my own confusion using github. This PR is correct and has all the desired commits of my other two which can be rejected.

MatusMak commented on index.js:
```diff
@@ -40,19 +46,36 @@ if (argv.dictionary) {
+  let attemptCounts = 0;
+  /**
+   * Token will expire when token timer reaches
+   * the max alloted time, in seconds.
```
```suggestion
   * the max allowed time, in seconds.
```

MatusMak commented on .eslintrc.json:
```diff
@@ -0,0 +1,15 @@
+{
```
Would you mind adding [eslint dev dependency](https://www.npmjs.com/package/eslint) and add a new npm script that will trigger lint by running `npm run lint` in the console?

mat888 replied:
Yeah I'll get to this.

MatusMak commented on index.js:
```diff
@@ -40,19 +46,36 @@ if (argv.dictionary) {
+   * Creates an object which calls checkFunction once
+   * a second. Is terminated with clearInterval(timerInterval);
+   */
+  const timerInterval = setInterval(checkToken, 1000);
```
I know that in the ticket I only mentioned that `setTimeout` shouldn't be used, but `setInterval` is the same idea here. I haven't try that but just judging from what I can see, this will call `checkToken` every second regardless of whether user entered an input or not, which is not a good behavior.

Instead, I suggest to do it without using of any sort of timer function and instead compare timestamp of when token was generated to the timestamp of when user attempted to guess the token.

MatusMak commented on index.js:
```diff
@@ -40,19 +46,36 @@ if (argv.dictionary) {
+    }
+
+    tokenTimer += 1;
+    if (tokenTimer >= expirationTime) {
```
Token validity should be checked before verifying it. With this code, if I enter correct token way after it expired, it would still pass.

MatusMak reviewed (changes requested):
Hey, I posted my review, if you have any questions or want to discuss something let me know! :)
